code/modules/photography/camera/camera.dm

```diff
@@ -251,11 +251,11 @@
 			to_chat(user, span_notice("[pictures_left] photos left."))
 
 		if(can_customise)
-			var/customise = tgui_alert(user, "Do you want to customize the photo?", "Customization", list("Yes", "No"))
+			var/customise = user.is_holding(new_photo) && tgui_alert(user, "Do you want to customize the photo?", "Customization", list("Yes", "No"))
 			if(customise == "Yes")
-				var/name1 = tgui_input_text(user, "Set a name for this photo, or leave blank.", "Name", max_length = 32)
-				var/desc1 = tgui_input_text(user, "Set a description to add to photo, or leave blank.", "Description", max_length = 128)
-				var/caption = tgui_input_text(user, "Set a caption for this photo, or leave blank.", "Caption", max_length = 256)
+				var/name1 = user.is_holding(new_photo) && tgui_input_text(user, "Set a name for this photo, or leave blank.", "Name", max_length = 32)
+				var/desc1 = user.is_holding(new_photo) && tgui_input_text(user, "Set a description to add to photo, or leave blank.", "Description", max_length = 128)
+				var/caption = user.is_holding(new_photo) && tgui_input_text(user, "Set a caption for this photo, or leave blank.", "Caption", max_length = 256)
```

Comment:
I was confused for a second because I was certain that this would be returning booleans but then I re-read the ref and saw `The return value is equal to the last argument to be evaluated.` and that's pretty cool beans. i don't really know how i feel about the pattern though but the only solution i have is to make the code quite a bit longer so i'll chalk it up to "whatever"

Reply:
Short circuiting yes